experimental/dds/tree2/src/simple-tree/proxies.ts

```diff
@@ -253,7 +254,11 @@ const listPrototypeProperties: PropertyDescriptorMap = {
 			this: TreeListNode<AllowedTypes, "javaScript">,
 			index: number,
 		): FlexTreeUnknownUnboxed | undefined {
-			return getSequenceField(this).at(index);
+			const val = getSequenceField(this).at(index);
+			if (val === undefined || val === null || isPrimitiveValue(val) || isFluidHandle(val)) {
```

Comment:
Thanks for jumping into a new part of the code Connor! It's a big help to have more pairs of eyes over here. 

I think this is correct behavior, however it diverges a little from the general pattern that the proxies use to access children. In the other cases (e.g. `list[number]` - see `maybeNumber` below), they retrieve the _boxed_ item from the lower flex API (even when it is a leaf) and then pass it directly to `getOrCreateNodeProxy`. There is an `atBoxed` in addition to `at` on the lower API.

It's not as simple as switching the `at` here to `atBoxed` though, because `atBoxed` doesn't accept negative indices (that might be a good test case to add in your unit test, btw).

You could update `boxedAt` to accept negative indices (not hard, but will required factoring out the code in the layer below that handles negative indices and using it in two places). I think that'd be an impromevent to the lower level API regardless of this code needing it; the signature for `boxedAt` should match `at`. Then, simply call `boxedAt` here and pass it to `getOrCreateNodeProxy`. The other reason this approach is good is because when you have the "boxed" item, you can ask its schema if it is a leaf (this is what `getOrCreateProxy` does). You can't do this here with the current approach because you are using the unboxed method (`at`). The leaves you get unbox automatically to the primitives. Thus, you have to do all these checks to see if it's a primitive, which are easier to get wrong and more brittle (e.g. it would break if we added a new kind of leaf, like handle2, and didn't update this here to include it).

If you're enjoying learning in this space, I'd recommend trying out my suggestion above. If you have other more pressing work to do, you can check this in as-is (you _have_ fixed the bug after all), and make a little followup tech-debt task to make `boxedAt` accept negative integers. In either case, please add a negative index check to the unit test.